src/main/java/io/numaproj/numaflow/sinker/OnSuccessMessage.java

```diff
@@ -0,0 +1,16 @@
+package io.numaproj.numaflow.sinker;
+
+import lombok.Builder;
+import lombok.Getter;
+
+import java.util.HashMap;
+
+@Getter
+@Builder
+public class OnSuccessMessage {
+    private final byte[] value;
+    private final String key;
+    private final HashMap<String, KeyValueGroup> userMetadata;
```

Comment:
Could you help me understand why we need to pass this in the response? Please comment on what is OnSuccessMessage and what each field represents.